src/Servers/Kestrel/Core/src/Internal/Http3/Http3FrameWriter.cs

```diff
@@ -280,9 +281,12 @@ public ValueTask<FlushResult> WriteResponseTrailersAsync(long streamId, HttpResp
                     var done = QPackHeaderWriter.BeginEncode(_headersEnumerator, buffer, ref _headersTotalSize, out var payloadLength);
                     FinishWritingHeaders(payloadLength, done);
                 }
-                catch (QPackEncodingException ex)
+                // Any exception from the QPack encoder can leave the dynamic table in a corrupt state.
+                // Since we allow custom header encoders we don't know what type of exceptions to expect.
+                catch (Exception ex)
                 {
                     _log.QPackEncodingError(_connectionId, streamId, ex);
+                    _connectionContext.Abort(new ConnectionAbortedException(ex.Message, ex));
                     _http3Stream.Abort(new ConnectionAbortedException(ex.Message, ex), Http3ErrorCode.InternalError);
```

Comment:
Shouldn't aborting the `_connectionContext` abort the stream implicitly?

Reply:
@JamesNK I wasn't sure about that part, I know state is tracked much differently from HTTP/2.